src/build/app/mod.rs

```diff
@@ -1481,40 +1470,160 @@ impl<'b> App<'b> {
-    fn _app_debug_asserts(&self) -> bool {
-        debugln!("App::_app_debug_asserts;");
-        for name in self.args.args.iter().map(|x| x.id) {
-            if self.args.args.iter().filter(|x| x.id == name).count() > 1 {
-                panic!(format!(
-                    "Arg names must be unique, found '{}' more than once",
-                    name
-                ));
+    #[allow(clippy::cognitive_complexity)]
+    fn _debug_asserts(&self) -> bool {
+        debugln!("App::_debug_asserts;");
+
+        for arg in &self.args.args {
+            debug_assert!(arg._debug_asserts());
+
+            // Name conflicts
+            assert!(
+                self.args.args.iter().filter(|x| x.id == arg.id).count() < 2,
+                "Argument name must be unique\n\n\t'{}' is already in use",
+                arg.name,
+            );
+
+            // Long conflicts
+            if let Some(l) = arg.long {
+                assert!(
+                    self.args.args.iter().filter(|x| x.long == Some(l)).count() < 2,
+                    "Argument long must be unique\n\n\t'--{}' is already in use",
+                    l
+                );
+            }
+
+            // Short conflicts
+            if let Some(s) = arg.short {
+                assert!(
+                    self.args.args.iter().filter(|x| x.short == Some(s)).count() < 2,
+                    "Argument short must be unique\n\n\t'-{}' is already in use",
+                    s
+                );
+            }
+
+            // Index conflicts
+            if let Some(idx) = arg.index {
+                assert!(
+                    positionals!(self).fold(0, |acc, p| if p.index == Some(idx) {
+                        acc + 1
+                    } else {
+                        acc
+                    }) < 2,
+                    "Argument '{}' has the same index as another positional argument\n\n\t \
+                    Use Arg::setting(ArgSettings::MultipleValues) to allow one \
+                    positional argument to take multiple values",
+                    arg.name
+                );
+            }
+
+            // requires, r_if, r_unless
+            if let Some(reqs) = &arg.requires {
+                for req in reqs {
+                    assert!(
+                        self.args.args.iter().any(|x| x.id == req.1)
+                            || self.groups.iter().any(|x| x.id == req.1),
+                        "Argument or group specified in 'requires*' for '{}' does not exist",
+                        arg.name,
+                    );
+                }
+            }
+
+            if let Some(reqs) = &arg.r_ifs {
+                for req in reqs {
+                    assert!(
+                        self.args.args.iter().any(|x| x.id == req.0)
+                            || self.groups.iter().any(|x| x.id == req.0),
+                        "Argument or group specified in 'required_if*' for '{}' does not exist",
```

Comment:
It would be better to also display the group's name

Reply:
Can't do it because `pub(crate) r_ifs: Option<Vec<(Id, &'help str)>>,`, which means the argument name is `Id` and not string.